src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -616,9 +619,24 @@ bool MetaspaceShared::link_class_for_cds(InstanceKlass* ik, TRAPS) {
   return res;
 }
 
+void MetaspaceShared::regenerate_lambdaforminvokers_holders(TRAPS) {
+  log_info(cds)("Regenerate lambdaform holder classes ...");
+  LambdaFormInvokers::regenerate_holder_classes(THREAD);
+  if (HAS_PENDING_EXCEPTION) {
+    log_info(cds)("%s: %s", PENDING_EXCEPTION->klass()->external_name(),
+                 java_lang_String::as_utf8_string(java_lang_Throwable::message(PENDING_EXCEPTION)));
+    log_info(cds)("Regenerate lambdaform holder classes ...failed");
+    CLEAR_PENDING_EXCEPTION; // Exceptions are ignored.
```

Comment:
As we discussed in the [previous PR](https://github.com/openjdk/jdk/pull/3611#discussion_r623578292), we will get an exception here if we hit an OOM, or we have generated a bad classfile. In either case, the exception cannot be just ignored. It should be reported to the user and the dynamic dump should fail.

I think we don't need this function anymore. Instead, the logging like `log_info(cds)("Regenerate lambdaform holder classes ...");` can be moved into `regenerate_holder_classes()`. The  `HAS_PENDING_EXCEPTION` check should be removed.